Source/Dafny/Printer.cs

```diff
@@ -385,27 +395,22 @@ public static string ExprToString(Expression expr)
       }
     }
 
-    private void PrintWitnessClause(RedirectingTypeDecl dd, int indent) {
+    private void PrintWitnessClause(RedirectingTypeDecl dd) {
       Contract.Requires(dd != null);
-      Contract.Requires(0 <= indent);
-      if (dd.WitnessKind == SubsetTypeDecl.WKind.None) {
-        return;
-      }
-      Indent(indent);
+      Contract.Requires(dd.WitnessKind != SubsetTypeDecl.WKind.None);
+
       switch (dd.WitnessKind) {
         case SubsetTypeDecl.WKind.Ghost:
           wr.Write("ghost ");
           goto case SubsetTypeDecl.WKind.Compiled;
         case SubsetTypeDecl.WKind.Compiled:
           wr.Write("witness ");
           PrintExpression(dd.Witness, true);
-          wr.WriteLine();
-          break;
-        case SubsetTypeDecl.WKind.None:
           break;
         case SubsetTypeDecl.WKind.Special:
-          wr.WriteLine("/*special witness*/");
+          wr.Write("/*special witness*/");
           break;
+        case SubsetTypeDecl.WKind.None:
```

Comment:
This is dead code given the contract, right?  If you want to leave it in for exhaustiveness, is there something like `impossible` or `die` we could use instead of just ignoring it?

Reply:
Correct. The precondition contract says it's not allowed, so this is dead code. Nevertheless, the coding pattern

        case SubsetTypeDecl.WKind.None:
        default:
          Contract.Assert(false);  // unexpected WKind
          break;

is used to say, in the source, that the developer thought about `case ...None`. By also throwing in `default` in that handling, the assertion will also trip in case other values of the enumeration are added in the future.

The `Assert(false)` has the effect of a `die`.